src/pipert2/core/base/routine.py

```diff
@@ -1,5 +1,180 @@
+from src.pipert2.core.handlers.message_handler import MessageHandler
+from src.pipert2.core.base.logger import PipeLogger
+from src.pipert2.utils.annotations import marking_functions_annotation
+from src.pipert2.utils.dummy_object import Dummy
+
+import threading
+import multiprocessing as mp
+from typing import Callable
 from abc import ABC, abstractmethod
+from functools import partial
 
 
 class Routine(ABC):
-    pass
+    """A routine is responsible for performing one of the component’s main tasks.
+    It can be run either as a thread or as a process. First it runs a setup function,
+    then it runs its main logic function in a continuous loop (until it is told to terminate),
+    and finally it runs a cleanup function.
```

Comment:
```suggestion
    """A routine is responsible for performing one of the flow’s main tasks.
    It can run as either a thread or a process. 
    First it runs a setup function, then it runs its main logic function in a continuous loop, until it is told to terminate.
    Once terminated it runs a cleanup function.
```